compiler-plugin/src/main/kotlin/arrow/meta/Meta.kt

```diff
@@ -8,18 +8,88 @@ import org.jetbrains.kotlin.com.intellij.mock.MockProject
 import org.jetbrains.kotlin.compiler.plugin.ComponentRegistrar
 import org.jetbrains.kotlin.config.CompilerConfiguration
 
+/**
+ * An Arrow Meta [Plugin] is a named function that given a [CompilerContext] produces a [List] of [ExtensionPhase].
+ *
+ * The following plugin named `"Hello World"` returns a single `func` extension phase that produces a transformation
+ * on the user tree.
+ *
+ * In this simple plugin we are replacing all functions named `helloWorld` with a new function that prints
+ * "Hello ΛRROW Meta!" when invoked.
+ *
+ * ```kotlin
+ * val Meta.helloWorld: Plugin get() =
+ *   "Hello World" {
+ *     meta(
+ *       func({ name == "helloWorld" }) { c ->  // <-- func(...) {...}
+ *         Transform.replace(
+ *           replacing = c,
+ *           newDeclaration = """|fun helloWorld(): Unit =
+ *                               |  println("Hello ΛRROW Meta!")
+ *                               |""".function.synthetic
+ *         )
+ *       }
+ *     )
+ *   }
+ * ```
+ *
+ * `func` is part of the [arrow.meta.quotes.Quote] DSL, a high level DSL built for tree transformations.
+ * While most use cases can be covered by the Quote DSL you can also subscribe to the low level compiler phases
+ * such as Configuration, Analysys, Resolution and Code generation with the Arrow Meta Compiler DSL [arrow.meta.dsl].
+ */
 data class Plugin(
   val name: String,
   val meta: CompilerContext.() -> List<ExtensionPhase>
 )
 
+/**
+ * Enables syntactic sugar for plugin creation via:
+ * "pluginId" {
+ *   meta(
+ *    ..phases
+ *   )
+ * }
+ */
```

Comment:
This is easily recognized as part of the DSL recipe - it might be good to mention this is the necessary component for the `plugin` since we're explaining what invoke does!